sdk/ml/azure-ai-ml/azure/ai/ml/entities/_job/pipeline/_pipeline_expression.py

```diff
@@ -542,9 +544,13 @@ def _generate_yaml_file(_path: Path) -> None:
             _data["outputs"]["output"]["type"] = self._result_type
             _command_inputs_items = []
             for _name in sorted(self._inputs):
-                _type = self._inputs[_name].type
-                _data["inputs"][_name] = {"type": _type}
-                _command_inputs_items.append(f"{_name}=\"$AZUREML_PARAMETER_{_name}\"")
+                _input = self._inputs[_name]
+                _data["inputs"][_name] = {"type": _input.type}
+                if isinstance(_input.value, NodeOutput):
+                    _param_name = f"{_input.value._owner.component.name}_{_input.value._name}"
+                else:
+                    _param_name = _name
+                _command_inputs_items.append(f"{_name}=\"$AZUREML_PARAMETER_{_param_name}\"")
             _command_inputs_string = " ".join(_command_inputs_items)
```

Comment:
need to figure out how command generate
may still need `mldesigner` to generate component
breaking case: <https://ml.azure.com/runs/loyal_dolphin_9lgtblq41t?wsid=/subscriptions/96aede12-2f73-41cb-b983-6d11a904839b/resourcegroups/sdk/workspaces/sdk-master>